scipy/fft/tests/test_basic.py

```diff
@@ -339,6 +339,24 @@ def test_dtypes_complex(self, dtype, xp):
         # Check both numerical results and exact dtype matches
         xp_assert_close(res_fft, x)
 
+    @skip_xp_backends(np_only=True,
+                      reasons=['array-likes only supported for NumPy backend'])
+    @pytest.mark.parametrize("op", [fft.fft, fft.ifft,
+                                    fft.fft2, fft.ifft2,
+                                    fft.fftn, fft.ifftn,
+                                    fft.rfft, fft.irfft,
+                                    fft.rfft2, fft.irfft2,
+                                    fft.rfftn, fft.irfftn,
+                                    fft.hfft, fft.ihfft,
+                                    fft.hfft2, fft.ihfft2,
+                                    fft.hfftn, fft.ihfftn,])
+    def test_array_like(self, xp, op):
+        x = [[[1.0, 1.0], [1.0, 1.0]],
+             [[1.0, 1.0], [1.0, 1.0]],
+             [[1.0, 1.0], [1.0, 1.0]]]
+        op(x) # should not error
```

Comment:
This seems like a better check

```suggestion
        xp_assert_close(op(x), op(xp.asarray(x)))
```

Reply:
Agreed. Apart from that, this branch seems array API test neutral on my local machine (no new issues) with GPU.

Note that you may have accidentally pushed your branch to the repo proper rather than your fork.

The diff is pretty clean, new test seems to run fast enough even with all the params.

Reply:
> Note that you may have accidentally pushed your branch to the repo proper rather than your fork.

Yep, I was using codespaces so it defaulted to that. Is that an issue, by the way? Sure, we wouldn't want a lot of branches cluttering the main repo, but are there any other issues with it?

Reply:
> but are there any other issues with it?

Not sure if there are any major issues with it, but my intuition is that an uptick in devs writing to the main repo branch space does at least slightly increased the risk of something annoying happening.